### PR TITLE
For #12554: Helper class for password reveal on logins

### PR DESCRIPTION
**Pull request by hakkikaancaliskan**
Rebase of #12564

https://imgur.com/ZVLvLhp

`showPassword` is unnecessary because as default we're hiding password and if user clicks, we're just toggling it to show.
This method will toggle the state. Hide -> Show, Show -> Hide.

cc @person808 

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [x] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/logins/view/LoginDetailView.kt [hunk 1]
```diff
@@ -4,62 +4,19 @@
...
-import kotlinx.android.synthetic.main.fragment_login_detail.view.*
-import org.mozilla.fenix.R
-import org.mozilla.fenix.components.metrics.Event
-import org.mozilla.fenix.ext.components
 import org.mozilla.fenix.settings.logins.LoginsListState
 
 /**
  * View that contains and configures the Login Details
  */
 @Suppress("ForbiddenComment")
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/settings/logins/view/LoginDetailView.kt:15]: nit -  we may be able to remove this suppress now (I think it was about the TODO)

File: app/src/main/java/org/mozilla/fenix/settings/logins/PasswordRevealHelper.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+fun togglePasswordReveal(
+    passwordText: TextView,
+    revealPasswordButton: ImageButton,
+    password: String? = null
+) {
+    val context = passwordText.context
+
+    // If password null, then it means we're showing in EditLoginFragment then if not null
+    // we're showing in LoginDetailFragment
+    val currText = password ?: passwordText.text
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/settings/logins/PasswordRevealHelper.kt:20]: Do we have to pass in the password here? It's unclear to me why we need this. Shouldn't the passwordText have the password in both cases?

File: app/src/main/java/org/mozilla/fenix/settings/logins/PasswordRevealHelper.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package org.mozilla.fenix.settings.logins
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/settings/logins/PasswordRevealHelper.kt:1]: nit license

File: app/src/main/java/org/mozilla/fenix/settings/logins/PasswordRevealHelper.kt (hunk 2 above)
Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/settings/logins/PasswordRevealHelper.kt:20]: Before last commit on EditLoginsFragment and LoginsDetailFragment they were not using same way for refreshing.
EditLogins was using passwordText.text and LoginsDetailFragment was using logins?.password.
So I didn't touch them but looks like with latest commit they are refreshing passwordText with same way.
I'm going to test and update pr if there is no problem.

Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/settings/logins/PasswordRevealHelper.kt]: Updated.

Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/settings/logins/PasswordRevealHelper.kt]: You can check with video in comment zero (updated video).

**Comment by ekager** [comment]
Opened as https://github.com/mozilla-mobile/fenix/pull/12674

**Pull request closed without merging**